base/bucket_test.go

```diff
@@ -358,44 +370,64 @@ func mockCertificatesAndKeys(t *testing.T) {
 	certBytes, err = x509.CreateCertificate(rand.Reader, &clientTemplate, &rootTemplate, &clientKey.PublicKey, rootKey)
 	assert.NoError(t, err, "Client certificate should be generated")
 	saveAsCertFile(t, clientCertPath, certBytes)
+
+	return certPaths
 }
 
 func saveAsKeyFile(t *testing.T, filename string, key *ecdsa.PrivateKey) {
 	file, err := os.Create(filename)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	b, err := x509.MarshalECPrivateKey(key)
-	assert.NoError(t, err)
+	require.NoError(t, err)
 	err = pem.Encode(file, &pem.Block{Type: "EC PRIVATE KEY", Bytes: b})
-	assert.NoError(t, err)
-	assert.NoError(t, file.Close())
+	require.NoError(t, err)
+	require.NoError(t, file.Close())
 }
 
 func saveAsCertFile(t *testing.T, filename string, derBytes []byte) {
 	certOut, err := os.Create(filename)
-	assert.NoError(t, err, "No error while opening cert.pem for writing")
+	require.NoError(t, err, "No error while opening cert.pem for writing")
 	err = pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: derBytes})
-	assert.NoError(t, err, "No error while writing data to cert.pem")
+	require.NoError(t, err, "No error while writing data to cert.pem")
 	err = certOut.Close()
-	assert.NoError(t, err, "No error while closing cert.pem")
+	require.NoError(t, err, "No error while closing cert.pem")
+}
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
 }
 
-func removeFiles(names []string) error {
-	for _, name := range names {
-		if err := os.Remove(name); err != nil {
-			return err
-		}
+func dirExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
 	}
-	return nil
+	return info.IsDir()
 }
 
 func TestTLSConfig(t *testing.T) {
 	// Mock fake root CA and client certificates for verification
-	mockCertificatesAndKeys(t)
+	certPaths := mockCertificatesAndKeys(t)
+	certPath := certPaths["certPath"]
+	rootKeyPath := certPaths["rootKeyPath"]
+	rootCertPath := certPaths["rootCertPath"]
+	clientKeyPath := certPaths["clientKeyPath"]
+	clientCertPath := certPaths["clientCertPath"]
+	log.Printf("certPaths: %s", certPaths)
+
+	require.True(t, fileExists(rootKeyPath), "File %v should exists", rootKeyPath)
+	require.True(t, fileExists(rootCertPath), "File %v should exists", rootCertPath)
+	require.True(t, fileExists(clientKeyPath), "File %v should exists", clientKeyPath)
+	require.True(t, fileExists(clientCertPath), "File %v should exists", clientCertPath)
```

Comment:
`mockCertificatesAndKeys` already has some assertions in it, so we could verify the file paths exist in there, before the function returns anything.

Reply:
Yeah, it would be better to verify the existence of file path inside mockCertificatesAndKeys; fixed in latest revision.